dll/win32/kernel32/client/appcache.c

```diff
@@ -139,20 +139,122 @@ IsShimInfrastructureDisabled(VOID)
  * @unimplemented
  */
 BOOL
-WINAPI
-BaseCheckAppcompatCache(IN PWCHAR ApplicationName,
-                        IN HANDLE FileHandle,
-                        IN PWCHAR Environment,
-                        OUT PULONG Reason)
+BasepShimCacheCheckBypass(
+    _In_ LPCWSTR ApplicationName,
```

Comment:
PCWSTR here and elsewhere in the other functions you've added.

Reply:
Used `PCWSTR`.